client/app/competitor/page.tsx

```diff
@@ -12,6 +13,37 @@ import {
 } from '@/components/ui/accordion';
 import { ScrollArea } from '@/components/ui/scroll-area';
 import { Separator } from '@/components/ui/separator';
+import Leaderboard from '../leaderboard/page';
+
+const ObserveLeaderboardTab = () => {
+    const [isLeaderboard, setIsLeaderboard] = useState(false);
+
+    useEffect(() => {
+        const handleTabChange = (value: string) => {
+            if (value === 'Leaderboard') {
+                setIsLeaderboard(true);
+            } else {
+                setIsLeaderboard(false);
+            }
+        };
+
+        tabChangeEmitter.on('tabChange', handleTabChange);
+
+        return () => {
+            tabChangeEmitter.off('tabChange', handleTabChange);
+        };
+    }, []);
```

Comment:
It would be preferable to have this be more generic (both to allow multiple tabs and so we might be able to reuse it in the host page) by using something like
```ts
const [selectedTab, setSelectedTab] = useState<'text-editor' | 'leaderboard'>('text-editor');
```

and then have the condition below check against that value.